draft-ietf-quic-transport.md

```diff
@@ -4099,11 +4102,13 @@ preferred_address (0x000d):
   the end of the handshake, as described in {{preferred-address}}.  The format
   of this transport parameter is the PreferredAddress struct shown in
   {{fig-preffered-address}}.  This transport parameter is only sent by a server.
+  Servers MAY choose to only send a preferred address of one address family by
+  sending a zero-length address for the other family.
 
 ~~~
    struct {
-     enum { IPv4(4), IPv6(6), (15) } ipVersion;
-     opaque ipAddress<4..2^8-1>;
+     opaque ipv4Address<0..2^8-1>;
```

Comment:
Aren't these two fields now fixed sizes?  If so, any reason not to specify those?

Reply:
Yep.  I think that the invalid address (0.0.0.0 or ::) is probably OK here.

Reply:
IIUC the text says that the value should be a zero-length byte array if the address type is not used, rather than using an invalid address.

Therefore I think the correct notation is:
```
opaque ipv4Address<0..4>;
opaque ipv6Address<0..16>;
```

Reply:
@kazuho suggestion SGTM.